R/objects.R

```diff
@@ -1525,6 +1585,66 @@ as.Graph.matrix <- function(x, ...) {
   return(as.Graph.Matrix(x = as(object = x, Class = 'Matrix')))
 }
 
+#' @param counts name of the CellDataSet assay to store as \code{counts}
+#' @param data name of the CellDataSet assay to slot as \code{data}
+#'
+#' @rdname as.Seurat
+#' @export
+#' @method as.Seurat CellDataSet
+#'
+as.Seurat.CellDataSet <- function(x, counts = "counts", ...) {
+  if (!PackageCheck('monocle', error = FALSE)) {
+    stop("Please install monocle from Bioconductor before converting to a CellDataSet object")
+  }
+  counts <- tryCatch(
+    expr = exprs(x),
+    error = function(e) {
+      stop("No data in provided assay - ", counts)
+    }
+  )
+  meta.data <- as.data.frame(x = pData(x))
+  # if cell names are NULL, fill with cell_X
+  if (is.null(x = colnames(x = counts))) {
+    warning("The column names of the 'counts' and 'data' matrices are NULL. Setting cell names to cell_columnidx (e.g 'cell_1').")
+    cell.names <- paste0("cell_", 1:ncol(x = counts))
+    colnames(x = counts) <- cell.names
+    rownames(x = meta.data) <- cell.names
+  }
+  # Creating the object
+  seurat.object <- CreateSeuratObject(counts = counts, meta.data = meta.data)
+  # Adding dim reductions if there are any
+  for (dr in slotNames(x = x)) {
+    # reasonable expectation that this corresponds to dim red
+    if (typeof(x = slot(object = x, name = dr)) == 'double') {
+      embeddings <- slot(object = x, name = dr)
+      embeddings <- as.data.frame(x = embeddings)
+      if (dim(x = embeddings)[2] == length(x = cell.names)) {
+        # Flip so columns are coordinates
+        embeddings <- t(x = embeddings)
+        # Make sure the cell/row names match.
+        # rownames(x = embeddings) <- cell.names
+        rownames(x = embeddings) <- colnames(x = seurat.object)
+        key <- gsub(
+          pattern = "[[:digit:]]",
+          replacement = "_",
+          x = colnames(x = x@auxClusteringData[[dr]])[1]
```

Comment:
Pleased use `slot` or a Monocle-provided accessor function instead of the `@` sigil.